src/core/server/saved_objects/import/collect_saved_objects.ts

```diff
@@ -79,7 +81,12 @@ export async function collectSavedObjects({
     }),
     createFilterStream<SavedObject>((obj) => (filter ? filter(obj) : true)),
     createMapStream((obj: SavedObject) => {
-      importIdMap.set(`${obj.type}:${obj.id}`, {});
+      if (dataSourceId) {
+        importIdMap.set(`${dataSourceId}_${obj.type}:${obj.id}`, {});
```

Comment:
This doesn't look right, can you check the logic and make sure it functions as expected? Key should be ${obj.type}:${dataSourceId}_${obj.id}, right?

Reply:
why do we want to attach the datasource id before the type?